quarkus/dist/src/main/content/bin/kc.sh

```diff
@@ -85,7 +85,11 @@ fi
 # Specify options to pass to the Java VM.
 #
 if [ "x$JAVA_OPTS" = "x" ]; then
-   JAVA_OPTS="-Xms64m -Xmx512m -XX:MetaspaceSize=96M -XX:MaxMetaspaceSize=256m -Djava.net.preferIPv4Stack=true -Dfile.encoding=UTF-8 -Dsun.stdout.encoding=UTF-8 -Dsun.err.encoding=UTF-8 -Dstdout.encoding=UTF-8 -Dstderr.encoding=UTF-8 -XX:+ExitOnOutOfMemoryError -Djava.security.egd=file:/dev/urandom"
+   # The defaults set up Keycloak with '-XX:+UseParallelGC -XX:MinHeapFreeRatio=10 -XX:MaxHeapFreeRatio=20 -XX:GCTimeRatio=4 -XX:AdaptiveSizePolicyWeight=90' which proved to provide a good throughput and efficiency in the total memory allocation and CPU overhead.
+   # If the memory is not used, it will be freed. See https://developers.redhat.com/blog/2017/04/04/openjdk-and-containers for details.
+   # To optimize for large heap sizes or for throughput and better response time due to shorter GC pauses, consider ZGC and Shenandoah GC.
+   # Both ZGC and Shenandoah GC seem to be more eager to claim the maximum heap size. Tests showed that ZGC might need additional tuning as as it is not as aggressive as ParallelGC in reclaiming dead objects.
```

Comment:
@vmuzikar - this inline docs follow the best practices of capturing the "why" of some decision which was taken in the code someone else or me in the future can figure out how we came up with these choices as they are non-obvious by just looking at them. Therefore I'd like to keep them. 

To override the settings, people can `JAVA_OPTS_APPEND`. Those will override the existing settings as needed, so there is no need to override the whole `JAVA_OPTS`. 

This PR is about gaining parity with RHSSO 7.6 performance. To add a full JVM Heap and GC guide, this would IMHO call for a separate issue. If we want to make them switchable, this would also require additional smoke testing to ensure those configurations actually work.

Reply:
Thanks for the context. I agree with capturing the motivation "for future generations". :) Just the wording of it gave me (if I put myself in a user's shoes) the impression that we actually encourage and support using of alternative GC. But it's not a blocker for me, we can leave as is as it is a subjective matter.

If I'm not mistaken `JAVA_OPTS_APPEND` just appends additional input, does not selectively override/replace provided options. That means that you can potentially get two GC configurations in the resulting command which behaviour might be undefined.

Reply:
`JAVA_OPTS_APPEND` would append the additional input, that's true. At least for the memory options, I know that the one that is at the end of the options overrides the one that is there earlier - this is the way we for example configure heap sizes in keycloak-benchmark. 

Once we support different garbage collectors, I agree this should be handled differently. I've created a follow-up ticket #20780 for this. Please update the description as needed.

Reply:
Thanks for creating the follow-up issue, works for me.